src/string.cpp

```diff
@@ -358,6 +358,33 @@ void StrTrimInPlace(std::string &str)
 	StrRightTrimInPlace(str);
 }
 
+/**
+ * Check whether the given string starts with the given prefix.
+ * @param str    The string to look at.
+ * @param prefix The prefix to look for.
+ * @return True iff the begin of the string is the same as the prefix.
+ */
+bool StrStartsWith(const std::string_view str, const std::string_view prefix)
+{
+	size_t prefix_len = prefix.size();
+	if (str.size() < prefix_len) return false;
+	return str.compare(0, prefix_len, prefix, 0, prefix_len) == 0;
```

Comment:
I am staring at this, and think: sure? I really have no clue if this is doing the right thing :D

We really need unit-tests for these kind of things .. :P

Reply:
Or even better, use library quality code such as https://en.cppreference.com/w/cpp/string/basic_string_view/starts_with

If the comment is about the compare, cppreference says "`constexpr int compare(size_type pos1, size_type count1, basic_string_view v, size_type pos2, size_type count2) const;` (3)
...
 3) Equivalent to `substr(pos1, count1).compare(v.substr(pos2, count2))`."

Reply:
No, the comment is not about the compare. The comment is that I have no clue if this is doing the right thing under all circumstance, as that is near impossible to figure out based on the code. For that, unit-tests would be very welcome.

Reply:
Right, as mentioned, reviewing this by code was really difficult to estimate if this would do the right thing. So I made a mini-unit-text:

https://godbolt.org/z/bxErvPY3o

It seems to be doing what I expect, but let me know if I forgot any test-cases.